provider/redis.go

```diff
@@ -94,49 +135,211 @@ func (store *redisOnlineStore) DeleteTable(feature, variant string) error {
 	return nil
 }
 
+func (store *redisOnlineStore) CreateIndex(feature, variant, entity string, vectorType VectorType) (VectorStoreTable, error) {
+	key := redisTableKey{store.prefix, feature, variant}
+	cmd := store.createIndexCmd(key, entity, vectorType)
+	resp := store.client.Do(context.Background(), cmd)
+	if resp.Error() != nil {
+		return &redisOnlineIndex{}, resp.Error()
+	}
+	table := &redisOnlineIndex{client: store.client, key: key, valueType: vectorType}
+	return table, nil
+}
+
+// TODO: write unit tests for command creation
+func (store *redisOnlineStore) createIndexCmd(key redisTableKey, entity string, vectorType VectorType) rueidis.Completed {
+	requiredParams := []string{
+		"TYPE", "FLOAT32",
+		"DIM", strconv.FormatUint(uint64(vectorType.Dimension), 10),
+		"DISTANCE_METRIC", "COSINE",
+	}
+	return store.client.B().
+		FtCreate().
+		Index(key.String()).
+		Schema().
+		FieldName(vectorField).
+		Vector("HNSW", int64(len(requiredParams)), requiredParams...).
+		Build()
+}
+
+func (store *redisOnlineStore) GetIndex(feature, variant string) (string, error) {
+	return "", nil
```

Comment:
?

Reply:
Unimplemented as of yet. I have question about if we want to use a separate naming convention for the index key and if we want a different struct to represent the index (e.g. `algo: "HNSW"` etc.)